pkg/common/api/error.go

```diff
@@ -88,3 +89,12 @@ func NewEndpointNotFound(msg string, args ...any) *ErrorResponse {
 		StatusCode: http.StatusNotFound,
 	}
 }
+
+// NewResourceAccessForbiddenError creates new Response object with ErrorAccessForbiddenError.
+func NewResourceAccessForbiddenError(msg string, args ...any) *ErrorResponse {
```

Comment:
Perhaps we should prefer to just use 404 for these errors, to avoid leaking information?